debug/org.eclipse.debug.ui/ui/org/eclipse/debug/internal/ui/DebugUIPlugin.java

```diff
@@ -372,6 +372,22 @@ public static Shell getShell() {
 		return null;
 	}
 
+	/**
+	 * Returns shell that can be used to get the best parenting possible for a modal
+	 * dialog, or, if there is no one, currently active workbench window shell or
+	 * <code>null</code> if none.
+	 *
+	 * @return the modal shell or currently active workbench window shell or
+	 *         <code>null</code>
+	 */
+	public static Shell getShellForModalDialog() {
+		Shell shell = PlatformUI.getWorkbench().getModalDialogShellProvider().getShell();
+		if (shell == null) {
+			shell = getShell();
```

Comment:
Why is this needed? Seems to me that the (default) IShellProvider `ProgressManagerUtil::getDefaultParent` returns a shell if there is one. If it returns null, there is no shell, and the `DebugUIPlugin` won't find any either.

Reply:
Yes, it can return null. My point was: If it returns null, then `getShell()` won't find anything either.